Modules/_hashopenssl.c

```diff
@@ -329,24 +345,37 @@ EVP_repr(EVPobject *self)
     return PyUnicode_FromFormat("<%U HASH object @ %p>", self->name, self);
 }
 
-#if HASH_OBJ_CONSTRUCTOR
+/*[clinic input]
+_hashlib.HASH.__init__ as EVP_tp_init
+
+    name as name_obj: object
```

Comment:
Both here and in `EVP_new`, the input string is converted into a `char *` via `PyArg_Parse(name_obj, "s", &name)`, but also used as-is later. Using `unicode` rather than `object` seems unnecessary since the `PyArg_Parse` does a similar, more restrictive check of the input. So, for now, I've kept the parameter types as `object`.

It is likely possible to use `unicode` as you suggest, and replace the `PyArg_Parse` calls with something simpler that assumes the input is a unicode object, perhaps `PyUnicode_AsASCIIString()` or `PyUnicode_AsUTF8()`.

What do you recommend?

Reply:
We need to check that the input string doesn't contain null characters, since it is used as null-terminated string. `PyArg_Parse("s")` performs this check in contrary to simpler methods.

I agree that it is better to keep object for now. I see an opportunity for small refactoring, but it can be made in other PR.